hyperspy/models/model2D.py

```diff
@@ -0,0 +1,268 @@
+    existing components as a template.
+
+    Once defined, the model can be fitted to the data using :meth:`fit` or
+    :meth:`multifit`. Once the optimizer reaches the convergence criteria or
+    the maximum number of iterations the new value of the component parameters
+    are stored in the components.
+
+    It is possible to access the components in the model by their name or by
+    the index in the model. An example is given at the end of this docstring.
+
+    Note that methods are not yet defined for plotting 2D models or using
+    gradient based optimisation methods - these will be added soon.
+
+    Attributes
+    ----------
+
+    image : Image instance
+        It contains the data to fit.
+    chisq : A Signal of floats
+        Chi-squared of the signal (or np.nan if not yet fit)
+    dof : A Signal of integers
+        Degrees of freedom of the signal (0 if not yet fit)
+    red_chisq : Signal instance
+        Reduced chi-squared.
+    components : `ModelComponents` instance
+        The components of the model are attributes of this class. This provides
+        a convinient way to access the model components when working in IPython
+        as it enables tab completion.
+
+    Methods
+    -------
+
+    append
+        Append one component to the model.
+    extend
+        Append multiple components to the model.
+    remove
+        Remove component from model.
+    fit, multifit
+        Fit the model to the data at the current position or the full dataset.
+
+    See also
+    --------
+    Base Model
+    Model1D
+
+    Example
+    -------
+
+
+
+    """
+
+    def __init__(self, image, dictionary=None):
+        self.image = image
+        self.signal = self.image
+        self.axes_manager = self.signal.axes_manager
+        self.xaxis, self.yaxis = np.meshgrid(
+            self.axes_manager.signal_axes[0].axis,
+            self.axes_manager.signal_axes[1].axis)
+        self.axes_manager.connect(self.fetch_stored_values)
+        self.free_parameters_boundaries = None
+        self.channel_switches = np.ones(self.xaxis.shape, dtype=bool)
+        # self._position_widgets = []
+        self._plot = None
+        self.chisq = image._get_navigation_signal()
+        self.chisq.change_dtype("float")
+        self.chisq.data.fill(np.nan)
+        self.chisq.metadata.General.title = self.signal.metadata.General.title + \
+            ' chi-squared'
+        self.dof = self.chisq._deepcopy_with_new_data(
+            np.zeros_like(
+                self.chisq.data,
+                dtype='int'))
+        self.dof.metadata.General.title = self.signal.metadata.General.title + \
+            ' degrees of freedom'
+        # self._suspend_update = False
+        self._adjust_position_all = None
+        self._plot_components = False
+        self.components = ModelComponents(self)
+        if dictionary is not None:
+            self._load_dictionary(dictionary)
```

Comment:
ok -- I think I removed that because I thought it was only used by ModelSpecialSlicers -- which I put in Model1D but not Model2D.

Looking at the code for ModelSpecialSlicers though it is probably dimension ambiguous if defined for .signal rather than .spectrum - do you think that's right? If so shall I put that in BaseModel then?